7001DataCleaning.Rmd

```diff
@@ -177,57 +178,217 @@ clean$paint_color <- impute(clean$paint_color,'unknown')
+title_status <- title_status[title_status!='missing']
+for (i in li_descr){
+  for (j in title_status){
+    if (str_detect(i,j)==T){
+      df.unknown[df.unknown$description==i,]$title_status <- j
+    }
+  }
+}
+clean[clean$title_status=='missing',]$title_status <- df.unknown$title_status
+clean['title_status'][clean['title_status'] == 'missing'] <- 'unknown'
+```
 
-```{r}
-glimpse(data)
+6. transmission
+
+```{r extract transmission}
+df.unknown = clean[clean$transmission=='unknown',]
+li_descr = as.list(df.unknown$description)
+transmission <- unique(clean$transmission)
+transmission <- transmission[transmission!='unknown' & transmission!='other']
+for (i in li_descr){
+  for (j in transmission){
+    if (str_detect(i,j)==T){
+      df.unknown[df.unknown$description==i,]$transmission <- j
+    }
+  }
+}
+clean[clean$transmission=='unknown',]$transmission <- df.unknown$transmission
+```
+
+7. drive
+
+```{r extract drive}
+df.unknown = clean[clean$drive=='unknown',]
+li_descr = as.list(df.unknown$description)
+drive <- unique(clean$drive)
+drive <- drive[drive!='unknown']
+for (i in li_descr){
+  for (j in drive){
+    if (str_detect(i,j)==T){
+      df.unknown[df.unknown$description==i,]$drive <- j
+    }
+  }
+}
+clean[clean$drive=='unknown',]$drive <- df.unknown$drive
+```
+
+8. size
+
+```{r extract size}
+df.unknown = clean[clean$size=='unknown',]
+li_descr = as.list(df.unknown$description)
+size <- unique(clean$size)
+size <- size[size!='unknown']
+for (i in li_descr){
+  for (j in size){
+    if (str_detect(i,j)==T){
+      df.unknown[df.unknown$description==i,]$size <- j
+    }
+  }
+}
+clean[clean$size=='unknown',]$size <- df.unknown$size
+```
+
+9. type
+
+```{r extract type}
+df.unknown = clean[clean$type=='unknown',]
+li_descr = as.list(df.unknown$description)
+type <- unique(clean$type)
+type <- type[type!='unknown' & type!='other']
+for (i in li_descr){
+  for (j in type){
+    if (str_detect(i,j)==T){
+      df.unknown[df.unknown$description==i,]$type <- j
+    }
+  }
+}
+clean[clean$type=='unknown',]$type <- df.unknown$type
 ```
 
-## Data Type Conversion
 
-#### Convert type of variables
+10. paint_color
+
+```{r extract paint_color}
+df.unknown = clean[clean$paint_color=='unknown',]
+li_descr = as.list(df.unknown$description)
+paint_color <- unique(clean$paint_color)
+paint_color <- paint_color[paint_color!='unknown']
+for (i in li_descr){
+  for (j in paint_color){
+    if (str_detect(i,j)==T){
+      df.unknown[df.unknown$description==i,]$paint_color <- j
+    }
+  }
+}
+clean[clean$paint_color=='unknown',] <- df.unknown
+```
+
+#### drop description column
+
+```{r drop descr}
+clean = select(clean,-description)
+```
+
+#### add make_model column
+
+```{r}
+clean['make_model'] <- paste(clean$manufacturer,clean$model,sep=':')
+```
 
-```{r convert-to-chr-type}
 
-data$id <- as.character(data$id)
+#### Output clean data
 
+```{r}
+write.csv(clean, "clean-vehicles.csv", row.names = F)
 ```
 
-```{r convert-to-factor-type}
```

Comment:
what is part for? and why is deleted?

Reply:
it's for converting string vars to factor type, I delete because when I re-read csv, the vars type will change back to string. I find a method to use read.csv('xxx.csv',stringAsFactors=TRUE) to read the file, and the str var will automatically change to factor